backend/GameManager.py

```diff
@@ -0,0 +1,53 @@
+# import pygame as pg
+from backend.Board import Board
+from backend.Piece import Pieces, Turn
+
+class GameManager:
+    def __init__(self, size, pieces):
+        self.board = Board(size)
+        self.pieces = Pieces(pieces)
+        self.turn = Turn.WHITE
+
+        self.all_possible_moves = self.board.get_valid_moves()
+        self.current_moves = self.all_possible_moves
+
+    def get_board(self, row=-1, col=-1):
+        if row == -1 or col == -1:
+            return self.board
+        return self.board.check_position(row, col)
+
+    def get_turn(self):
+        if self.turn == Turn.WHITE:
+            return 'white'
+        else:
+            return 'black'
+    
+    def placement_complete(self):
+        return self.pieces.all_pieces_placed()
+
+    def piece_placement(self, row, column):
+        if (row, column) in self.current_moves:
+            if self.board.check_position(row, column) != "empty":
+                print(self.board.check_position(row, column).lower())
+                return "GameManagerError -- position not empty"
+            
+            flag = 0
```

Comment:
We could name this isPiecedPlaced or something similar to capture what the variable represents

Reply:
May even want to throw exception instead of return `False` in the set_piece functions then you can try-except it here, then we could do something on screen to communicate that it was invalid position without changing how the set_piece function returns values.

Reply:
That was something I was going to ask about, but we didn't meet with everyone after I did Pieces and honestly I didn't think it mattered that much for this first sprint: we never discussed how we wanted to do error handling. The way I have it set up now is easy to find and change, so we can absolutely do that, but I do think we should discuss it at some point.